src/ts/address-search.ts

```diff
@@ -391,42 +388,54 @@ export default class AddressValidation {
       }
 
       // Regex that checks if the input is the format for a what3words search. Ex: ///a.b.c
-      const regex = /^\/{0,}(?:[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+|[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3})$/;
+      var regex = /^\/{0,}(?:[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+|[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3})$/;
 
-      if (regex.test(this.currentSearchTerm.trim()) && this.options.enableWhat3Words && this.what3WordCountries.indexOf(this.currentCountryCode) > -1) {
-        this.isWhat3Words = true;
+      if (regex.test(this.currentSearchTerm.trim())) {
+        this.avMode = AddressValidationMode.WHAT3WORDS;
         this.currentSearchTerm = this.currentSearchTerm.trim();
       }
-      else {
-        this.isWhat3Words = false;
+
+      // is UPRN or UDPRN
+      regex = /^\d{12}|\d{8}$/;
+      if (regex.test(this.currentSearchTerm.trim())) {
+        this.avMode = AddressValidationMode.UDPRN;
+        this.currentSearchTerm = this.currentSearchTerm.trim();
       }
 
       // Fire an event before a search takes place
       this.events.trigger('pre-search', this.currentSearchTerm);
 
-      // Construct the new Search URL and data
-      const data = this.isWhat3Words ? this.generateLookupDataForApiCall(this.currentSearchTerm, true) : this.generateSearchDataForApiCall();
-
       // Store the last search term
       this.lastSearchTerm = this.currentSearchTerm;
 
-      // Hide the inline search spinner
-      this.searchSpinner.hide();

```

Comment:
Not sure it's necessary to remove this line? If the spinner isn't hidden first, then it might result in multiple spinners!

Reply:
Added it back in and added a comment why we should leave it in